### ISSUE TITLE
XML filenames for BLAST?

### ISSUE THREAD

**Issue by sdwfrost** [rega-cev/rega-genotype#7]
If I try to run the genotype tool 'as is', I get the following error:

```
Exception in thread "main" java.io.FileNotFoundException: /Users/sdf22/Programs/rega-genotype/rega-genotype/xml/NoV/blast.xml (No such file or directory)
```

I have to copy/rename the BLAST config in the NoV directory to get this to work.

**Comment by vagnerfonseca** [comment]
you need to edit the "global-conf.sample.xml" file located in the folder etc, putting the commands of each software.

**Comment by sdwfrost** [comment]
All the commands of the software are in place; it looks as though the filenames within the xml directory are hardcoded - am I wrong? It works if I rename the BLAST config to `blast.xml` (present for some organisms but not others). Similarly, the SELF analysis dies unless there's a `config.xml` present in the xml/organism subdirectory.

**Comment by Ewout1988** [comment]
That is correct. We are currently in a process of generalising the tool for a lot of different viruses/pathogens.
Before this tool was created for each virus specifically (I think we had HIV, Norovirus and Enterovirus in the beginning).
Each virus has its specific xml folder. Those which contain the files: blast.xml, blast.fasta, config.xml, 'organism'.fasta, genome_0.png, phylo-'organism'.xml and resources.xml are ready to be used by the generic version of GenotypeTool.
The reason why we haven't updated HIV, NoV and ENV is because those viruses are a bit more complex than e.g. zika virus (recombination, ...) so we can't bring them to the generic tool yet. Thus changing the filenames for those 3 viruses won't necessarily give you the correct typing results as there are some steps hardcoded in the code which are not present in the generic tool. It is required that you call the specific Java classes which are located under rega.genotype.ui.viruses for that.
